Rns 4.5.33 (#29848)
openshift/openshift-docs: release_notes/ocp-4-5-release-notes.adoc

[id="ocp-4-5-33-bug-fixes"]
==== Bug fixes

* Previously, an incorrect OVN-kubernetes security rule blocked certain inbound connections. While rare, some attempts to connect to a pod would fail spuriously. This update fixes the iptables to block the intended connections, which results in no spurious failures. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1921283[*BZ#1921283*])

* Previously, a watch cache in the Kubernetes API was initialized from the global revision (`etcd`) and could remain for an undefined period if no changes were made. This behavior sometimes led to a situation in which a client got a resource version (RV) from a server that had observed a newer RV, disconnected from it due to a network error, and reconnected to a server that was behind, resulting in `Timeout: Too large resource version` errors. With this release, the reflector is fixed so that it can recover from these errors, and Operators that use the `client-go` library for getting notifications from the server can recover and make progress upon receiving the errors. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1877346[*BZ#1877346*])

> * Previously, attempts to write to a nil writer may have led to `invalid memory address` or `nil pointer dereference` errors. Sharing the same instance of the writer might have also led to an `index out of range [43] with length 30 and recovered from err index > windowEnd` error. This update fixes a data race in the `SerializeObject` function of the kube-apiserver. link:https://bugzilla.redhat.com/show_bug.cgi?id=1879208[*BZ#1879208*]

Reviewer: ```suggestion
* Previously, attempts to write to a nil writer might have led to `invalid memory address` or `nil pointer dereference` errors. Sharing the same instance of the writer might have also led to an `index out of range [43] with length 30 and recovered from err index > windowEnd` error. This update fixes a data race in the `SerializeObject` function of the kube-apiserver. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1879208[*BZ#1879208*])
```

Reviewer: ISG recommends using "might" or "can" over "may".

> * Previously, a misplaced array index while pruning records from memory would cause excessive memory usage and fail to remove old reports from the archive. This update changes the array index key so that prune successfully removes records from memory without causing excessive memory usage. link:https://bugzilla.redhat.com/show_bug.cgi?id=1894243[*BZ#1894243*]

Reviewer: ```suggestion
* Previously, a misplaced array index while pruning records from memory would cause excessive memory usage and fail to remove old reports from the archive. This update changes the array index key so that prune successfully removes records from memory without causing excessive memory usage. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1894243[*BZ#1894243*])
```

> * Previously, the Red Hat Linux CoreOS (RHCOS) was using a stage repo location for the kernel-rt package. Consequently, the kernel-rt package would not synchronize to the vanilla kernel package. This update changes the RHCOS build config to use the production repo location, which properly syncs the kernel-rt package with the vanilla kernel package.  link:https://bugzilla.redhat.com/show_bug.cgi?id=1922262[*BZ#1922262*]

Reviewer: ```suggestion
* Previously, the {op-system-first} was using a stage repo location for the kernel-rt package. Consequently, the kernel-rt package would not synchronize to the vanilla kernel package. This update changes the {op-system} build config to use the production repo location, which properly syncs the kernel-rt package with the vanilla kernel package. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1922262[*BZ#1922262*])
```

Reviewer: See the [common-attributes](https://raw.githubusercontent.com/openshift/openshift-docs/master/modules/common-attributes.adoc) file for a list of replaceable tokens. These more or less just have to be memorized for when to use them :sweat_smile:

Author: Thanks!
